pkg/scheduler/schedule_one.go

```diff
@@ -92,9 +92,10 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	schedulingCycleCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	scheduleResult, assumedPodInfo, err := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, start, podsToActivate)
-	if err != nil {
-		sched.FailureHandler(schedulingCycleCtx, fwk, assumedPodInfo, err, scheduleResult.reason, scheduleResult.nominatingInfo, start)
+	scheduleResult, assumedPodInfo, status := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, start, podsToActivate)
+	if !status.IsSuccess() {
+		status.AppendReason(scheduleResult.reason)
```

Comment:
1) I think we no longer need the reason any more, we can tell via the status.  So remove the reason from `ScheduleResult`. 
```pseudo-code
reason := PodReasonSchedulerError
if status.IsUnscehdulable {
    reason = PodReasonUnschedulable
}
```

2) no need to append reason then